src/BoilerPlate/App.hpp

```diff
@@ -5,10 +5,18 @@
 // C++ STL
 #include <string>
 #include <list>
+#include <iostream>
+#include <algorithm>
+#include <GL/glew.h>
+#include <SDL2/SDL_opengl.h>
```

Comment:
Only add headers when needed. Also try to use forward declarations instead of including stuff in the header just to make the compiler happy.